.github/workflows/build.yml

```diff
@@ -0,0 +1,34 @@
+name: Build
+
+on:
+  push:
+    branches:
+      - main
+  pull_request:
```

Comment:
This event will be handled in a separate workflow for pull request which will be different than the main workflow (For example, there will be no deployment in the PRs workflow)